bin/reth/src/cli/ext.rs

```diff
@@ -130,7 +131,7 @@ pub trait RethNodeCommandConfig: fmt::Debug {
         &mut self,
         conf: &Conf,
         components: &Reth,
-    ) -> eyre::Result<PayloadBuilderHandle>
+    ) -> eyre::Result<PayloadBuilderHandle<BuiltPayload>>
```

Comment:
I am missing only here to use a generic Payload instead of the BuiltPayload struct.

My problem is that since this is a default trait implementation, I am struggling a bit to constrain in this scenario this BuiltPayload.

```rust
    fn spawn_payload_builder_service<Conf, Reth, Payload>(
        &mut self,
        conf: &Conf,
        components: &Reth,
    ) -> eyre::Result<PayloadBuilderHandle<Payload>>
```

Would then cause an error on the return type as we constrain the PayloadGenerator to create Jobs for BuiltPayload.

Reply:
this is currently not possible, we need this to support generics on this level

https://github.com/paradigmxyz/reth/issues/5501